universe/interface.go

```diff
@@ -1190,3 +1191,46 @@ type Telemetry interface {
 	QueryAssetStatsPerDay(ctx context.Context,
 		q GroupedStatsQuery) ([]*GroupedStats, error)
 }
+
+// AuthenticatedIgnoreTuple wraps the existing SignedIgnoreTuple struct and
+// includes information that allows it to be authenticated against an ignore
+// tree universe root.
+//
+// TODO(roasbeef): supplement with bitcoin header proof
+type AuthenticatedIgnoreTuple struct {
+	*SignedIgnoreTuple
+
+	// IgnoreTreeRoot is the root of the ignore tree that the ignore tuple
+	// resides within.
+	IgnoreTreeRoot mssmt.Node
+
+	// InclusionProof is the universe inclusion proof for the ignore tuple
+	// within the universe tree.
+	InclusionProof *mssmt.Proof
+}
+
+// TupleQueryResp is the response to a query for ignore tuples.
+type TupleQueryResp = lfn.Result[lfn.Option[[]AuthenticatedIgnoreTuple]]
```

Comment:
Why include `lfn.Option` here? Wouldn't an empty slice signal the same thing as a `None`?

I think a similar sort of question can be asked for `SumQueryResp` below. I assume that if there are any leaves in the tree then the sum is always `> 0`, am I wrong? In which case `None` and `0` mean the same thing.

Reply:
I went back and forth a bit here: the `None` value makes it explicit when the query has no response value, while the empty slice is an implicit indicator. 

> I assume that if there are any leaves in the tree then the sum is always > 0, am I wrong? In which case None and 0 mean the same thing.

Same thing here re implicit vs explicit. If there's no tree, then a sum value isn't defined. An implicit interpretaiton is that if there's no tree, then the sum is zero.